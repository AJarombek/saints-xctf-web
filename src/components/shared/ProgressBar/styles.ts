/**
 * JSS styles for the ProgressBar component.
 * @author Andrew Jarombek
 * @since 2/11/2021
 */

import Colors from '../../../styles/colors';

type StyleProps = {
  progress: number;
  inProgress: boolean;
};

export default {
  progressBar: {
    position: 'relative'
  },
  bar: {
    position: 'absolute',
    height: 16,
    margin: '4px 0'
  },
  barFull: {
    backgroundColor: Colors.lightBackground,
    width: '100%'
  },
  barProgress: {
    backgroundColor: (props: StyleProps): string => (props.progress >= 100 ? Colors.sxctfRed : Colors.spotPaletteBrown),
    width: (props: StyleProps): string => `${Math.min(props.progress, 100)}`,
    backgroundImage: (props: StyleProps): string =>
      props.inProgress
        ? `
        linear-gradient(
          -45deg,
          rgba(255,255,255,.15) 25%, transparent 25%,
          transparent 50%, rgba(255,255,255,.15) 50%,
          rgba(255,255,255,.15) 75%, transparent 75%,
          transparent
        )
    `
        : 'none',
    backgroundSize: '2rem'
  }
};
