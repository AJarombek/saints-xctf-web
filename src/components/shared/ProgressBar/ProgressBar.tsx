/**
 * ProgressBar component.
 * @author Andrew Jarombek
 * @since 2/11/2021
 */

import React from 'react';
import { createUseStyles } from 'react-jss';
import styles from './styles';
import classNames from 'classnames';

interface Props {
  progress: number;
  inProgress: boolean;
}

const useStyles = createUseStyles(styles);

const ProgressBar: React.FunctionComponent<Props> = ({ progress, inProgress }) => {
  const classes = useStyles({ progress, inProgress });

  return (
    <div className={classes.progressBar}>
      <div className={classNames(classes.bar, classes.barFull)} />
      <div className={classNames(classes.bar, classes.barProgress)} />
    </div>
  );
};

export default ProgressBar;
