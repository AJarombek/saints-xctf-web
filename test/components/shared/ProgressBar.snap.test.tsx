/**
 * Snapshot test for the PaginationBar component.
 * @author Andrew Jarombek
 * @since 3/7/2021
 */

import React from 'react';
import renderer from 'react-test-renderer';
import ProgressBar from '../../../src/components/shared/ProgressBar';

describe('PaginationBar Snapshot Tests', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ProgressBar progress={50} inProgress={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
