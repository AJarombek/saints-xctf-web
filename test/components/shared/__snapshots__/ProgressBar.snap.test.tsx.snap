// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationBar Snapshot Tests renders correctly 1`] = `
<div
  className="progressBar-0-1-1"
>
  <div
    className="bar-0-1-2 barFull-0-1-3"
  />
  <div
    className="bar-0-1-2 barProgress-0-1-4 barProgress-d0-0-1-5"
  />
</div>
`;
